### ISSUE TITLE
[Question] Getting an access token for a managed identity on a VM to call a custom web API scope

### ISSUE THREAD

**Issue by MoodieG** [Azure/azure-sdk-for-python#35951]
Hey,

I am trying to get an access token to call a custom web API scope and I am getting `AADSTS500011` error that the resource principal was not found in the tenant.

Steps to reproduce:

- Create an app registration in Azure with Web authentication enabled and ID tokens is checked.
- Add a scope in the Expose an API section
- Create a managed identity resource in Azure
- Create a VM resource in Azure
- Assigned the managed identity as user assigned on the VM
- Connect to the VM
- Install python on the VM
- Install azure.identity module on the VM. Example: `pip install azure.identity`
- Run the below python script:

```
from azure.identity import DefaultAzureCredential

# Set the client ID of your API app
api_client_id = "app-registration-client-id"

scope_name = "my-scope"

# Set the scope for the custom API
scope = f"api://{api_client_id}/{scope_name}"

# Create a DefaultAzureCredential
credential = DefaultAzureCredential()

# Get an access token
token = credential.get_token(scope)

# Print the access token
print(f"Access token for {scope}: {token.token}")
```

See the following error:

`AADSTS500011: The resource principal named api://app-registration-client-id/my-scope was not found in the tenant`

Any idea why this is not working? For reference the `.default` scope works.

**Comment by xiangyan99** [comment]
Thanks for reaching out.

Because it is a user-assigned managed identity, you need to specify the client id when you create the credential ojbect: https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/identity/azure-identity/azure/identity/_credentials/default.py#L71

Please let us know if it does not work.

**Comment by MoodieG** [comment]
/unresolve 

Thank you for your prompt response. I have tried specifying the managed identity client id by updating the above script to:

`credential = DefaultAzureCredential(managed_identity_client_id="{MyClientID}")` 

And I tried setting up a user environment variable with name `AZURE_CLIENT_ID` but got the following error:

`ManagedIdentityCredential: ManagedIdentityCredential authentication unavailable. The requested identity has not been assigned to this resource. Error: Unexpected response "{'error': 'invalid_request', 'error_description': 'Identity not found'}"`

Is there something else I need to do to assigned a managed identity to a VM? I just following the documentation here:

https://learn.microsoft.com/en-us/entra/identity/managed-identities-azure-resources/how-to-configure-managed-identities?pivots=qs-configure-portal-windows-vm#assign-a-user-assigned-managed-identity-to-an-existing-vm
